### PR TITLE
Fix maven Pom warnings

### PR DESCRIPTION
**Pull request by timtay-microsoft**
Removes duplicate groupId designations (when not specified, groupId is inherited), and suppresses a warning about a duplicate dependency declaration

### CODE REVIEW

File: device/iot-device-samples/pom.xml [hunk 1]
```diff
@@ -40,6 +40,11 @@
         <module>custom-sas-token-provider-sample</module>
     </modules>
     <dependencies>
+        <!--suppress MavenDuplicateDependenciesInspection -->
+        <!--
+        This duplicate dependency declaration warning can be ignored as one child Pom.xml duplicates this dependency declaration
+        demonstrate how to exclude in order to demonstrate excluding bouncycastle dependency from the device client library.
```
Comment (abhipsaMisra) [inline device/iot-device-samples/pom.xml:46]: I think there is a typo here: "...duplicates this dependency declaration ~demonstrate how to exclude~ in order to demonstrate excluding bouncycastle dependency"

**Review by abhipsaMisra — APPROVED** [review]
So we now have the group Id defined at the parent level of each project, and everyone else under it inherits it, correct?

**Comment by timtay-microsoft** [comment]
> So we now have the group Id defined at the parent level of each project, and everyone else under it inherits it, correct?

That's the idea, yep!

**Pull request merged**